lib/FreeRTOS-Plus-TCP/source/FreeRTOS_UDP_IP.c

```diff
@@ -347,6 +347,19 @@ UDPPacket_t *pxUDPPacket = (UDPPacket_t *) pxNetworkBuffer->pucEthernetBuffer;
 		/* There is no socket listening to the target port, but still it might
 		be for this node. */
 
+		#if( ipconfigUSE_DNS == 1 ) && ( ipconfigDNS_USE_CALLBACKS == 1 )
```

Comment:
Wouldn't this proposed code make it easier for the device to be spoofed? A rogue device on the network can send DNS packets to the device at any time and the device will add those entries to its cache.